mem/log.go

```diff
@@ -124,7 +125,20 @@ func (log *memlog) Append(v interface{}) (margaret.Seq, error) {
 	log.l.Lock()
 	defer log.l.Unlock()
 
-	log.tail.next = &memlogElem{v: v, seq: log.tail.seq + 1, wait: make(chan struct{})}
+	nxt := &memlogElem{
+		v:    v,
+		seq:  log.tail.seq + 1,
+		wait: make(chan struct{}),
+	}
+
+	// scroll to prev
```

Comment:
So we are looking for the _second_ last here? Why don't we keep track of that in `log.tail`? That way we don't have to seek through the entire thing every time.

I mean it's only the memory-backed implementation, but still...

Reply:
yup, this was a brainfart.

i already added the backlink pointer `prev` to the elements for seeking backwards.